test.py

```diff
@@ -268,52 +322,5 @@ def inference_args(parser, remaining_args):
     args, _ = parser.parse_known_args()
     args = inference_args(parser, args)
 
-    use_cuda = not args.no_cuda and torch.cuda.is_available()
-    device = torch.device("cuda" if use_cuda else "cpu")
-
-    for input_file in args.input:
-        # handling an input audio path
-        audio, rate = sf.read(input_file, always_2d=True)
-
-        if audio.shape[1] > 2:
-            warnings.warn(
-                'Channel count > 2! '
-                'Only the first two channels will be processed!')
-            audio = audio[:, :2]
-
-        if rate != args.samplerate:
-            # resample to model samplerate if needed
-            audio = resampy.resample(audio, rate, args.samplerate, axis=0)
-
-        if audio.shape[1] == 1:
-            # if we have mono, let's duplicate it
-            # as the input of OpenUnmix is always stereo
-            audio = np.repeat(audio, 2, axis=1)
-
-        estimates = separate(
-            audio,
-            targets=args.targets,
-            model_name=args.model,
-            niter=args.niter,
-            alpha=args.alpha,
-            softmask=args.softmask,
-            residual_model=args.residual_model,
-            device=device
-        )
-        if not args.outdir:
-            model_path = Path(args.model)
-            if not model_path.exists():
-                outdir = Path(Path(input_file).stem + '_' + args.model)
-            else:
-                outdir = Path(Path(input_file).stem + '_' + model_path.stem)
-        else:
-            outdir = Path(args.outdir)
-        outdir.mkdir(exist_ok=True, parents=True)
-
-        # write out estimates
-        for target, estimate in estimates.items():
-            sf.write(
-                outdir / Path(target).with_suffix('.wav'),
-                estimate,
-                args.samplerate
-            )
+    test_main(args.input, args.samplerate, args.niter, args.alpha, args.softmask, args.residual_model, args.model,
+              args.targets, args.outdir, args.no_cuda)
```

Comment:
use named arguments for readability reasons

Reply:
Done, hadn't done it before since the argument name was exactly the same except the `args.` part